glean-core/ffi/src/lib.rs

```diff
@@ -128,10 +141,50 @@ where
     with_glean_mut(|glean| Ok(callback(glean)))
 }
 
+struct FdLogger {
+    pub f: Option<RwLock<File>>,
+}
+
+#[derive(Serialize)]
+struct FdLoggingRecord {
+    level: String,
+    message: String,
+}
+
+static mut FD_LOGGER: FdLogger = FdLogger { f: None };
+
+impl log::Log for FdLogger {
+    fn enabled(&self, _metadata: &log::Metadata) -> bool {
+        true
+    }
+
+    fn log(&self, record: &log::Record) {
+        if self.enabled(record.metadata()) {
+            if let Some(file) = &self.f {
+                let payload = FdLoggingRecord {
+                    level: record.level().to_string(),
+                    message: record.args().to_string(),
+                };
+                let _ = writeln!(
+                    file.write().unwrap(),
+                    "{}",
+                    serde_json::to_string(&payload).unwrap()
+                );
+            }
+        }
+    }
+
+    fn flush(&self) {
+        if let Some(file) = &self.f {
+            let _ = file.write().unwrap().flush();
+        }
+    }
+}
+
 /// Initialize the logging system based on the target platform. This ensures
 /// that logging is shown when executing the Glean SDK unit tests.
 #[no_mangle]
-pub extern "C" fn glean_enable_logging() {
+pub extern "C" fn glean_enable_logging(fd: usize) {
```

Comment:
Maybe this should be `c_int` or `c_uint`? `usize`'s size depends on the platform (32 vs 64bit), using `c_int` it's almost always a 32bit integer and thus easier to match the types on the other side.

Reply:
It looks like a Windows file handle is a pointer, which is why I went with `usize`.  I guess this matters for JNA where you have to write something that will work everywhere (but not for Python cffi where the pointer size is known and used at runtime).  I guess the best choice is probably a `u64` (which will only contain 32-bits of meaningful stuff on i686).

Reply:
Ah, I see. 
At least in the _definition_ on the Kotlin side we could use [`IntegerType`](https://java-native-access.github.io/jna/4.2.0/com/sun/jna/IntegerType.html), which seems to be the equivalent?

Given that we don't pass in any value other than `0` from Kotlin it's less of an issue.
We should document properly what the `fd` should be if set and that `0` is treated as disabling fd logging.

Reply:
I guess `u64` is also fine. At least it's defined and consistent across platforms.